src/main/java/org/jabref/gui/mergeentries/MultiMergeEntriesView.java

```diff
@@ -0,0 +1,428 @@
+                                 TaskExecutor taskExecutor) {
+        this.preferences = preferences;
+        this.taskExecutor = taskExecutor;
+
+        viewModel = new MultiMergeEntriesViewModel();
+
+        ViewLoader.view(this)
+                  .load()
+                  .setAsDialogPane(this);
+
+        ButtonType mergeEntries = new ButtonType(Localization.lang("Merge entries"), ButtonBar.ButtonData.OK_DONE);
+        this.getDialogPane().getButtonTypes().setAll(ButtonType.CANCEL, mergeEntries);
+        this.setResultConverter(viewModel::resultConverter);
+
+        viewModel.entriesProperty().addListener((ListChangeListener<MultiMergeEntriesViewModel.Entry>) c -> {
+            while (c.next()) {
+                if (c.wasAdded()) {
+                    for (MultiMergeEntriesViewModel.Entry entryColumn : c.getAddedSubList()) {
+                        addColumn(entryColumn);
+                    }
+                }
+            }
+        });
+
+        viewModel.mergedEntryProperty().get().getFieldsObservable().addListener((MapChangeListener<Field, String>) change -> {
+            if (change.wasAdded() && !fieldRows.containsKey(change.getKey())) {
+                FieldRow fieldRow = new FieldRow(
+                        change.getKey(),
+                        viewModel.mergedEntryProperty().get().getFields().size() - 1);
+                fieldRows.put(change.getKey(), fieldRow);
+            }
+        });
+    }
+
+    @FXML
+    public void initialize() {
+        topScrollPane.hvalueProperty().bindBidirectional(centerScrollPane.hvalueProperty());
+        leftScrollPane.vvalueProperty().bindBidirectional(centerScrollPane.vvalueProperty());
+        rightScrollPane.vvalueProperty().bindBidirectional(centerScrollPane.vvalueProperty());
+
+        viewModel.failedSuppliersProperty().addListener((obs, oldValue, newValue) -> {
+            failedSuppliers.setText(viewModel.failedSuppliersProperty().get().isEmpty() ? "" : Localization.lang("Could not extract Metadata from: %0", viewModel.failedSuppliersProperty().stream().collect(Collectors.joining(", "))));
+        });
+
+        fillDiffModes();
+    }
+
+    private void fillDiffModes() {
+        diffMode.setItems(FXCollections.observableList(List.of(MergeEntries.DiffMode.PLAIN, MergeEntries.DiffMode.WORD, MergeEntries.DiffMode.CHARACTER)));
+        new ViewModelListCellFactory<MergeEntries.DiffMode>()
+                .withText(MergeEntries.DiffMode::getDisplayText)
+                .install(diffMode);
+        MergeEntries.DiffMode diffModePref = preferences.getMergeDiffMode()
+                                                          .flatMap(MergeEntries.DiffMode::parse)
+                                                          .orElse(MergeEntries.DiffMode.WORD);
+        diffMode.setValue(diffModePref);
+
+        EasyBind.subscribe(this.diffMode.valueProperty(), mode -> {
+            preferences.storeMergeDiffMode(mode.name());
+        });
```

Comment:
@koppor this is why I need the big preferences object.

Reply:
@DominikVoigt and me are looking into reworking preferences handling. Should be merged anyways to move forward and leave the preferences problem to a more general solution.